### PR TITLE
PHP: AbstractCall remove decode method

### PR DESCRIPTION
**Pull request by jeffreyqw**
remove method_exists($data, 'encode')) and method_exists($obj, $deserializeFunc)

@stanley-cheung

### CODE REVIEW

File: src/php/lib/Grpc/AbstractCall.php [hunk 1]
```diff
@@ -114,9 +114,7 @@ public function cancel()
     protected function _serializeMessage($data)
     {
         // Proto3 implementation
-        if (method_exists($data, 'encode')) {
-            return $data->encode();
-        } elseif (method_exists($data, 'serializeToString')) {
+       if (method_exists($data, 'serializeToString')) {
```
Comment (stanley-cheung) [inline src/php/lib/Grpc/AbstractCall.php:117]: I think we can remove this `if` clause too.

Comment (jeffreyqw) [reply src/php/lib/Grpc/AbstractCall.php]: ok

File: src/php/lib/Grpc/AbstractCall.php [hunk 2]
```diff
@@ -141,12 +134,7 @@ protected function _deserializeResponse($value)
...
             $obj = new $className();
-            if (method_exists($obj, $deserializeFunc)) {
-                $obj->$deserializeFunc($value);
-            } else {
-                $obj->mergeFromString($value);
-            }
-
+            $obj->mergeFromString($value);
             return $obj;
         }
```
Comment (stanley-cheung) [inline src/php/lib/Grpc/AbstractCall.php:140]: Can you also remove the next 2 lines 
```
        // Protobuf-PHP implementation
        return call_user_func($this->deserialize, $value);
```

Then this is parallel to the work you did up in `_serializeMessage`.

Then you can also remove the `if (is_array($this->deserialize)) {` wrap.

So the entire `_deserializeResponse` function should be something like

```
    protected function _deserializeResponse($value)
    {
        if ($value === null) {
            return;
        }

        list($className, $deserializeFunc) = $this->deserialize;
        $obj = new $className();
        $obj->mergeFromString($value);
        return $obj;
    }
```

Comment (jeffreyqw) [reply src/php/lib/Grpc/AbstractCall.php:140]: got it

**Pull request merged**